geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/string/AbstractStringIntegrationTest.java

```diff
@@ -26,141 +28,190 @@
   }
 
   @Test
   public void testDecr_withWrongType_shouldError() {
     String key = "hashKey";
-    jedis.hset(key, "field", "non-int value");
+    jedis1.hset(key, "field", "non-int value");
 
-    assertThatThrownBy(() -> jedis.decr(key))
+    assertThatThrownBy(() -> jedis1.decr(key))
         .isInstanceOf(JedisDataException.class)
         .hasMessageContaining(RedisConstants.ERROR_WRONG_TYPE);
   }
 
   @Test
   public void testDecr_decrementsPositiveIntegerValues() {
     String key = "key";
-    jedis.set(key, "10");
+    jedis1.set(key, "10");
 
-    assertThat(jedis.decr(key)).isEqualTo(9L);
-    assertThat(jedis.get(key)).isEqualTo("9");
+    assertThat(jedis1.decr(key)).isEqualTo(9L);
+    assertThat(jedis1.get(key)).isEqualTo("9");
   }
 
   @Test
   public void testDecr_returnsValueWhenDecrementingResultsInNegativeNumber() {
     String key = "key";
-    jedis.set(key, "0");
+    jedis1.set(key, "0");
+
+    assertThat(jedis1.decr(key)).isEqualTo(-1L);
+    assertThat(jedis1.get(key)).isEqualTo("-1");
+  }
+
+  @Test
+  public void testIncrBy_returnsErrorMessageForWrongNumberOfParameters() {
+    assertThatThrownBy(() -> jedis1.sendCommand(Protocol.Command.INCRBY))
+        .hasMessageContaining("ERR wrong number of arguments for 'incrby' command");
+    assertThatThrownBy(() -> jedis1.sendCommand(Protocol.Command.INCRBY, "1"))
+        .hasMessageContaining("ERR wrong number of arguments for 'incrby' command");
+    assertThatThrownBy(() -> jedis1.sendCommand(Protocol.Command.INCRBY, "1", "2", "3"))
+        .hasMessageContaining("ERR wrong number of arguments for 'incrby' command");
+  }
+
+  @Test
+  public void testIncrBy_failsWhenPerformedOnNonIntegerValue() {
+    jedis1.sadd("key", "member");
+    assertThatThrownBy(() -> jedis1.incrBy("key", 1))
+        .hasMessageContaining("WRONGTYPE Operation against a key holding the wrong kind of value");
+  }
 
-    assertThat(jedis.decr(key)).isEqualTo(-1L);
-    assertThat(jedis.get(key)).isEqualTo("-1");
+  @Test
+  public void testIncrBy_createsAndIncrementsNonExistentKey() {
+    assertThat(jedis1.incrBy("nonexistentkey", 1)).isEqualTo(1);
+    assertThat(jedis1.incrBy("otherNonexistentKey", -1)).isEqualTo(-1);
+  }
+
+  @Test
+  public void testConcurrentIncrBy_performsAllIncrBys() {
+    String key = "key";
+    AtomicInteger expectedValue = new AtomicInteger(0);
+
+    jedis1.set(key, "0");
+
+    new ConcurrentLoopingThreads(1000,
+        (i) -> {
+          int increment = ThreadLocalRandom.current().nextInt(-50, 50);
+          expectedValue.addAndGet(increment);
+          jedis1.incrBy(key, increment);
+        },
+        (i) -> {
+          int increment = ThreadLocalRandom.current().nextInt(-50, 50);
+          expectedValue.addAndGet(increment);
+          jedis2.incrBy(key, increment);
+        }).run();
+
+    assertThat(Integer.parseInt(jedis1.get(key))).isEqualTo(expectedValue.get());
   }
```

Comment:
There's actually an `AbstractIncrByIntegrationTest` file. Some of these tests could be merged into that file. Others are duplicates. A test that isn't in either file: testing the max and min 64 bit signed integer value (9,223,372,036,854,775,807 and -9223372036854775808, respectively).  We should test both trying to increment a key that is set above/below these values (so setting a key to`9223372036854775808` then trying to incrby 1 (which results in `ERR value is not an integer or out of range`)) as well as incrementing a key that is right at the max/min value   (so setting a key to `9223372036854775807` then trying to incrby 1 (which results in `ERR increment or decrement would overflow`))

Reply:
I just found that file as well. good idea to condense them into one. And I like that test so I'll definitely implement that